add_project_files

Nikita-bug opened the pull request:
Sprint_4_project

Nikita-bug committed:
add_project_files

pomegranate282922 commented on src/main/java/ru/yandex/BasePage.java:
```diff
@@ -0,0 +1,12 @@
+
+    protected static final String MAIN_PAGE_URL = "https://qa-scooter.praktikum-services.ru/";
+
+    public static WebDriver driver;
```
Почему статический?
Нужно исправить

pomegranate282922 commented on src/test/java/finalProjectTests/MainPageTest.java:
```diff
@@ -0,0 +1,61 @@
+package finalProjectTests;
```
Пакеты именуем строчными буквами без camelCase, _ и . 
нужно исправить

pomegranate282922 commented on src/test/java/finalProjectTests/MainPageTest.java:
```diff
@@ -0,0 +1,61 @@
+        for (int i = 0; i < expectedValues.size(); i++) {
+            assertEquals(expectedValues.get(i).getTitle(), actualValues.get(i).getTitle());
+            assertEquals(expectedValues.get(i).getDescription(), actualValues.get(i).getDescription());
+        }
```
Все проверки должны быть независимыми. Нужно примерить параметризацию
Нужно исправить

pomegranate282922 commented on src/test/java/finalProjectTests/MainPageTest.java:
```diff
@@ -0,0 +1,61 @@
+    }
+
+    @After
+    public void after() {
```
Общие начала и концы всех тестовых классов удобно вынести в общий класс предок 
Можно улучшить

pomegranate282922 commented on src/test/java/finalProjectTests/MainPageTest.java:
```diff
@@ -0,0 +1,61 @@
+        driver.manage().window().maximize();
+        BasePage.driver = driver;
+        expectedValues = new ArrayList<>();
+        expectedValues.add(new FAQElement("Сколько это стоит? И как оплатить?", "Сутки — 400 рублей. Оплата курьеру — наличными или картой."));
```
Лучше List.of 
Можно улучшить

pomegranate282922 commented on src/test/java/finalProjectTests/OrderPageTest.java:
```diff
@@ -0,0 +1,96 @@
+        return new Object[][]{
+                {"Николай", "Крутов", "Рязань,390048", "88888888888", "23.10.2024", "Комментарий 1", "Хотите оформить заказ?"},
+                {"Максим", "Иванов", "Москва,400055", "87777777777", "25.10.2024", "Комментарий 2", "Заказ оформлен"},
+        };
```
Отлично что применена параметризация

pomegranate282922 commented on src/test/java/finalProjectTests/OrderPageTest.java:
```diff
@@ -0,0 +1,96 @@
+
+
+    @Test
+    public void checkingOrderPath1() {
```
Не используем магические цифры в именах 
Нужно исправить

pomegranate282922 reviewed (commented):
Работа хорошая! Применена параметризация. 
замечания:
1. Драйвер это нестатическое поле в Page Object 
2. Применить параметризацию в тестах на faq 
3. Нейминг

Nikita-bug committed:
1.Переименовал пакеты
2.Переименовал методы
3.Использовал параметризацию при тестировании аккордиона
4.Создал protected статический драйвер в отдельном классе, чтобы не создавать его в каждом классе

pomegranate282922 commented on src/test/java/finalprojecttests/MainPageTest.java:
```diff
@@ -0,0 +1,79 @@
+        FAQElement expectedValues = new FAQElement(expectedTitle, expectedValue);
+        //Коллекция с фактическими значениями
+        List<FAQElement> actualValues = mainPage.getFAQList();
+        for (int i = 0; i < actualValues.size(); i++) {
```
Так нехорошо. Каждый раз ты получаешь весь список, хотя проверить нужно всего одну строку. 
можно улучшить

pomegranate282922 commented on src/test/java/finalprojecttests/MainPageTest.java:
```diff
@@ -0,0 +1,79 @@
+        List<FAQElement> actualValues = mainPage.getFAQList();
+        for (int i = 0; i < actualValues.size(); i++) {
+            if (actualValues.get(i).getTitle().equals(expectedValues.getTitle())) {
+                assertEquals(actualValues.get(i).getTitle(), actualValues.get(i).getTitle());
```
Перепроверить ассерт 
Нужно исправить

pomegranate282922 commented on src/main/java/ru/yandex/BasePage.java:
```diff
@@ -0,0 +1,12 @@
+
+    protected static final String MAIN_PAGE_URL = "https://qa-scooter.praktikum-services.ru/";
+
+    protected static WebDriver driver;
```
Зачем статик? Нужно убрать 
Нужно исправить

pomegranate282922 reviewed (approved):
Многое исправил, но не все( еще нужно доработать.

pomegranate282922 reviewed (commented):
Случайно заапрувила)

Nikita-bug committed:
Исправил assert

pomegranate282922 commented on src/test/java/finalprojecttests/MainPageTest.java:
```diff
@@ -66,8 +66,8 @@ public void accordionTest() {
             if (actualValues.get(i).getTitle().equals(expectedValues.getTitle())) {
-                assertEquals(actualValues.get(i).getTitle(), actualValues.get(i).getTitle());
-                assertEquals(actualValues.get(i).getDescription(), actualValues.get(i).getDescription());
+                assertEquals(actualValues.get(i).getTitle(), expectedValues.getTitle());
```
Сначала ожидаемое значение, потом актуальное 
Нужно исправить

pomegranate282922 reviewed (commented):
И все же опять не до конца. Прошу внимательнее.

Nikita-bug committed:
Прошу прощения, исправил

pomegranate282922 reviewed (approved):
Отлично, зачет!
